src/database/postgres.go

```diff
@@ -18,7 +19,14 @@ type PgxIface interface {
 	Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error)
 }
 
-func initializeDatabaseSchema(ctx context.Context, conn PgxIface) error {
+func InitializeDatabaseSchema(ctx context.Context, conn PgxIface) error {
+	if ctx == nil {
+		return errors.New("context is nil")
+	}
+
+	if conn == nil {
```

Comment:
Is this actually possible? https://go101.org/article/nil.html has some info on what can and cannot be nil in go

I'm wondering if we should be passing conn as a pointer to avoid making a potentially expensive copy, in which case we would want to keep the nil check.

Reply:
I did encounter nil connection once (as a bug in my testcase which I then fixed), so I added these checks. But yes, passing a pointer can be better.